src/Distributed/halo_communication.jl

```diff
@@ -0,0 +1,172 @@
+using KernelAbstractions: @kernel, @index, Event, MultiEvent
+
+import Oceananigans.BoundaryConditions:
+    fill_halo_regions!,
+    fill_west_halo!, fill_east_halo!, fill_south_halo!,
+    fill_north_halo!, fill_bottom_halo!, fill_top_halo!
+
+#####
+##### MPI tags for halo communication BCs
+#####
+
+sides  = (:west, :east, :south, :north, :top, :bottom)
+
+side_id = Dict(
+    :east => 1, :west => 2,
```

Comment:
Any reason why you pick easy to be 1 and not 2?  This is fine and I don't suggest that you change it but the ordering is dfferent from what I would have expected.

Reply:
Ah I always order them in my head as east, west, north, south, top, bottom since it's easier to say (rolls off the tongue).

But actually now that I think of it this isn't really consistent... It's different from the `sides` tuple haha.

Reply:
As long as we are consistent, that's what matters, and I think you have been.  No reason to make this cosmetic change, I don't think.